runtime/ast/access.go

```diff
@@ -20,38 +20,173 @@ package ast
 
 import (
 	"encoding/json"
+	"strings"
 
 	"github.com/onflow/cadence/runtime/errors"
 )
 
-//go:generate go run golang.org/x/tools/cmd/stringer -type=Access
+//go:generate go run golang.org/x/tools/cmd/stringer -type=PrimitiveAccess
 
-type Access uint
+type Access interface {
+	isAccess()
+	Keyword() string
+	Description() string
+	String() string
+	MarshalJSON() ([]byte, error)
+	IsLessPermissiveThan(Access) bool
+}
+
+type Separator = string
+
+const (
+	Disjunction Separator = " |"
```

Comment:
Is the space a typo?

Reply:
I do actually use it later on in one of the type checking PRs, so having it be a separate type is probably safer.

Reply:
In that case maybe it's best to keep the type definition, but remove the values, and define additional constants for the documents as described above and use them based on the constant